src/accelerate/tracking.py

```diff
@@ -398,9 +403,147 @@ def finish(self):
+        description (`str`):
+            An optional string that populates the description box of the run. 
+            If a run is being resumed, the description is set on the resumed run. 
+            If a new run is being created, the description is set on the new run.
+    """
+
+    name = "mlflow"
+    requires_logging_directory = True
+
+    def __init__(
+        self,
+        experiment_name: str = None,
+        logging_dir: Optional[Union[str, os.PathLike]] = ".",
+        run_id: Optional[str] = None,
+        tags: Optional[Union[Dict[str, Any], str]] = None,
+        nested_run: Optional[bool] = False,
+        run_name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+
+        self._MAX_PARAM_VAL_LENGTH = mlflow.utils.validation.MAX_PARAM_VAL_LENGTH
+        self._MAX_PARAMS_TAGS_PER_BATCH = mlflow.utils.validation.MAX_PARAMS_TAGS_PER_BATCH
+
+        experiment_name = os.getenv("MLFLOW_EXPERIMENT_NAME", experiment_name)
+        run_id = os.getenv("MLFLOW_RUN_ID", run_id)
+        tags = os.getenv("MLFLOW_TAGS", tags)
+        if isinstance(tags, str):
+            tags = json.loads(tags)
+
+        nested_run = os.getenv("MLFLOW_NESTED_RUN", nested_run)
+
+        experiment_id = mlflow.create_experiment(
+            name=experiment_name,
+            artifact_location=logging_dir,
+            tags=tags,
+        )
+
+        mlflow.start_run(
+            run_id=run_id,
+            experiment_id=experiment_id,
+            run_name=run_name,
+            nested=nested_run,
+            tags=tags,
+            description=description,
+        )
+
+        logger.debug(f"Initialized mlflow experiment {experiment_name}")
+        logger.debug(
+            "Make sure to log any initial configurations with `self.store_init_configuration` before training!"
+        )
+
+    @property
+    def tracker(self):
+        return mlflow.active_run()
+
+
+    def store_init_configuration(self, values: dict):
+        """
+        Logs `values` as hyperparameters for the run. Should be run at the beginning of your experiment.
+
+        Args:
+            values (`dict`):
+                Values to be stored as initial hyperparameters as key-value pairs.
+        """
+
+        for name, value in list(values.items()):
+            # internally, all values are converted to str in MLflow
+            if len(str(value)) > self._MAX_PARAM_VAL_LENGTH:
+                logger.warning(
+                    f'Trainer is attempting to log a value of "{value}" for key "{name}" as a parameter. MLflow\'s'
+                    f" log_param() only accepts values no longer than {self._MAX_PARAM_VAL_LENGTH} characters so we dropped this attribute."
+                )
+                del values[name]
+
+        values_list = list(values.items())
+
+        # MLflow cannot log more than 100 values in one go, so we have to split it
```

Comment:
@sgugger @muellerzr , mlflow in AzureML cannot log more than 100 params (it's fine outside AzureML). Should I add anything to notify the user if it detects being in AzureML and the number of params > 100?

Reply:
Hi @muellerzr,

How can we get through the error of having more than 100 params? Can we restrict the number of params we would like to log? I am stuck in this issue right now. Any help is greatly appreciated

Regards

Reply:
Right below that you can see the for loop that was implemented

Reply:
@muellerzr I am sorry if I am asking very question. I have seen the for loop however, how do I implement this into my azure notebook? Do I need to import the entire .py as a library and call the function? or just use the for loop within my notebook thats in Azure ML? Please help.

Reply:
@dks198 

I made this modified version that will save the training arguments to a json file. At the moment, there doesn't seem to be a way around the 100 param limit, so the params might need to be saved as artifacts instead.

https://gist.github.com/nbroad1881/dbd3083804bf3007de9b83a90dcf6428

Reply:
@nbroad1881, thanks for sharing this. Can you please help me know how to use this in Azure ML Studio? Do I need to clone this entire notebook of yours to Azure ML?

Reply:
You can download the gist with 
```
wget https://gist.github.com/nbroad1881/dbd3083804bf3007de9b83a90dcf6428/raw/85a4ffc9c80b11fb3e9e8c86942b5b9e0101d98c/mlflow_tracker.py
```

Then you can import it by using 

`from mlflow_tracker import MLFlowTracker`

Then you create the tracker (filling in with your details)
```python
mlflow_tracker = MLflowTracker(
            experiment_name=,
            logging_dir=,
            run_id=,
            tags=,
            nested_run=,
            run_name=,
            description=,
        )
```

and pass it to the Accelerator
```python
Accelerator(log_with=[mlflow_tracker])
accelerator.init_trackers(project_name, param_dict)
```